browser/ai_chat/ai_chat_settings_helper.cc

```diff
@@ -103,10 +134,68 @@ void AIChatSettingsHelper::GetManageUrl(GetManageUrlCallback callback) {
 #if defined(OFFICIAL_BUILD)
   std::string domain = brave_domains::GetServicesDomain(kAccountHostnamePart);
 #else
-  std::string domain = brave_domains::GetServicesDomain(kAccountHostnamePart, brave_domains::STAGING);
+  std::string domain = brave_domains::GetServicesDomain(kAccountHostnamePart,
+                                                        brave_domains::STAGING);
 #endif
 
-  std::move(callback).Run(base::StrCat({url::kHttpsScheme, url::kStandardSchemeSeparator, domain}));
+  std::move(callback).Run(
+      base::StrCat({url::kHttpsScheme, url::kStandardSchemeSeparator, domain}));
+}
+
+void AIChatSettingsHelper::GetCustomModels(GetCustomModelsCallback callback) {
+  std::vector<mojom::CustomModelPtr> custom_models = GetCustomModels();
+  std::move(callback).Run(std::move(custom_models));
+}
+
+void AIChatSettingsHelper::AddCustomModel(mojom::CustomModelPtr model,
+                                          AddCustomModelCallback callback) {
+  if (!IsEndpointUrlValid(model->endpoint)) {
+    std::move(callback).Run(mojom::OperationResult::InvalidUrl);
+    return;
+  }
+
+  model_service_->AddCustomModel(std::move(model));
+  std::move(callback).Run(mojom::OperationResult::Success);
+}
+
+void AIChatSettingsHelper::SaveCustomModel(uint32_t index,
+                                           mojom::CustomModelPtr model,
+                                           SaveCustomModelCallback callback) {
+  if (!IsEndpointUrlValid(model->endpoint)) {
+    std::move(callback).Run(mojom::OperationResult::InvalidUrl);
+    return;
+  }
+
+  model_service_->SaveCustomModel(index, std::move(model));
+  std::move(callback).Run(mojom::OperationResult::Success);
+}
+
+void AIChatSettingsHelper::DeleteCustomModel(uint32_t index) {
+  model_service_->DeleteCustomModel(index);
+}
+
+void AIChatSettingsHelper::ChangeDefaultModel(const std::string& model_key) {
+  model_service_->ChangeDefaultModel(model_key);
+}
+
+void AIChatSettingsHelper::SetClientPage(
+    mojo::PendingRemote<mojom::SettingsPage> page) {
+  client_page_.Bind(std::move(page));
+}
+
+bool AIChatSettingsHelper::IsEndpointUrlValid(const GURL& url) {
+  GURL gurl(url);
+
+  if (!gurl.is_valid()) {
+    return false;
+  }
+
+  // We dont allow http scheme for non localhost urls
+  if (!net::IsLocalhost(gurl) && gurl.SchemeIs(url::kHttpScheme)) {
```

Comment:
this would still allow other insecure schemes like `ws`. suggest changing this to:

`if (gurl.SchemeIs(url::kHttpsScheme) || (net::IsLocalhost(gurl) && gurl.SchemeIs(url::kHttpScheme))) { return true }` otherwise return false at the end of this function

Reply:
coincidentally i noticed @yrliou just implemented the same logic here: https://github.com/brave/brave-core/pull/24249/files#diff-8c7ff8f18576f845460369a895a4399ddf5abc4d12b66cd7c46c0e54632e104aR355-R358

i would suggest both PRs use a single utility function so it's not duplicated

Reply:
FYI I'm moving my util out from wallet to be shared.